rl4co/models/common/constructive/nonautoregressive/encoder.py

```diff
@@ -0,0 +1,179 @@
+        are largely independent.
+
+    Args:
+        env_name: Name of the environment used to initialize embeddings
+        embed_dim: Dimension of the node embeddings
+        num_layers: Number of layers in the encoder
+        init_embedding: Model to use for the initial embedding. If None, use the default embedding for the environment
+        edge_embedding: Model to use for the edge embedding. If None, use the default embedding for the environment
+        act_fn: The activation function to use in each GNNLayer, see https://pytorch.org/docs/stable/nn.functional.html#non-linear-activation-functions for available options. Defaults to 'silu'.
+        agg_fn: The aggregation function to use in each GNNLayer for pooling features. Options: 'add', 'mean', 'max'. Defaults to 'mean'.
+    """
+
+    def __init__(
+        self,
+        embed_dim: int = 64,
+        env_name: Union[str, RL4COEnvBase] = "tsp",
+        # TODO: pass network
+        init_embedding: Optional[nn.Module] = None,
+        edge_embedding: Optional[nn.Module] = None,
+        graph_network: Optional[nn.Module] = None,
+        heatmap_generator: Optional[nn.Module] = None,
+        num_layers_heatmap_generator: int = 5,
+        num_layers_graph_encoder: int = 15,
+        act_fn="silu",
+        agg_fn="mean",
+        linear_bias: bool = True,
+    ):
+        super(NonAutoregressiveEncoder, self).__init__()
+        self.env_name = env_name.name if isinstance(env_name, RL4COEnvBase) else env_name
+
+        self.init_embedding = (
+            env_init_embedding(self.env_name, {"embed_dim": embed_dim})
+            if init_embedding is None
+            else init_embedding
+        )
+
+        self.edge_embedding = (
+            env_edge_embedding(self.env_name, {"embed_dim": embed_dim})
+            if edge_embedding is None
+            else edge_embedding
+        )
+
+        self.graph_network = (
+            GNNEncoder(
+                embed_dim=embed_dim,
+                num_layers=num_layers_graph_encoder,
+                act_fn=act_fn,
+                agg_fn=agg_fn,
+            )
+            if graph_network is None
+            else graph_network
+        )
+
+        self.heatmap_generator = (
+            EdgeHeatmapGenerator(
+                embed_dim=embed_dim,
+                num_layers=num_layers_heatmap_generator,
+                linear_bias=linear_bias,
+            )
+            if heatmap_generator is None
+            else heatmap_generator
+        )
+
+    def forward(self, td: TensorDict):
+        """Forward pass of the encoder.
+        Transform the input TensorDict into the latent representation.
+        """
+        # Transfer to embedding space
+        node_embed = self.init_embedding(td)
+        graph = self.edge_embedding(td, node_embed)
+
+        # Process embedding into graph
+        # TODO: standardize?
```

Comment:
Which boolean parameter do you mean?

Reply:
Just a boolean parameter `standardize` and then do ìf standardize ... else ...`, essentially only standardizing if the parameter is set to true. I ask if it's optional because there is a question mark in the comment :)